cloudflare/data_source_access_identity_provider.go

```diff
@@ -0,0 +1,82 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceCloudflareAccessIdentityProvider() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceCloudflareAccessIdentityProviderRead,
+
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ExactlyOneOf: []string{"zone_id", "account_id"},
+			},
+			"zone_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ExactlyOneOf: []string{"zone_id", "account_id"},
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceCloudflareAccessIdentityProviderRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	identifier, err := initIdentifier(d)
+	name := d.Get("name").(string)
+	if err != nil {
+		return err
+	}
+
+	providers := []cloudflare.AccessIdentityProvider{}
+	if identifier.Type == AccountType {
+		providers, err = client.AccessIdentityProviders(context.Background(), identifier.Value)
+	} else {
+		providers, err = client.ZoneLevelAccessIdentityProviders(context.Background(), identifier.Value)
+	}
+	if err != nil {
+		return fmt.Errorf("error listing Access Identity Providers: %s", err)
+	}
+
+	if len(providers) == 0 {
+		return fmt.Errorf("no Access Identity Provider found")
+	}
+
+	var accessIdentityProvider cloudflare.AccessIdentityProvider
+	for _, provider := range providers {
+		if provider.Name == name {
+			accessIdentityProvider = provider
+			break
+		}
+	}
+
+	if accessIdentityProvider.ID == "" {
+		return fmt.Errorf("no Access Identity Provider matching name '%s'", name)
```

Comment:
`%q` quotes the string value for you

```suggestion
		return fmt.Errorf("no Access Identity Provider matching name %q", name)
```